superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricPopoverTrigger.tsx

```diff
@@ -76,6 +76,26 @@ class AdhocMetricPopoverTrigger extends React.PureComponent<
     };
   }
 
+  componentDidUpdate(
+    prevProps: Readonly<AdhocMetricPopoverTriggerProps>,
+    prevState: Readonly<AdhocMetricPopoverTriggerState>,
+    snapshot?: any,
+  ) {
```

Comment:
```suggestion
  componentDidUpdate(prevProps: Readonly<AdhocMetricPopoverTriggerProps>) {
```

Is not the case for [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops)?

Reply:
Good point, I changed that. That required saving adhocMetric in state though, so that I could compare previous and next in getDerviedStateFromProps. Can you take another look?